crypto/encode_decode/decoder_pkey.c

```diff
@@ -576,11 +586,7 @@ ossl_decoder_ctx_for_pkey_dup(OSSL_DECODER_CTX *src,
 
     return dest;
  err:
-    if (process_data_dest != NULL) {
-        OPENSSL_free(process_data_dest->propq);
-        sk_EVP_KEYMGMT_pop_free(process_data_dest->keymgmts, EVP_KEYMGMT_free);
-        OPENSSL_free(process_data_dest);
-    }
+    decoder_clean_pkey_construct_arg(process_data_dest);
```

Comment:
This function additionally frees the object_type (It should be ok?)

Reply:
This is supposed to be a direct replacement of the original duplicated code. Do you see a difference?

Reply:
That free is needed if the field were used, so the cleanup is also a bug fix.

Reply:
object_type was the difference.. decoder_pkey.c sets it..